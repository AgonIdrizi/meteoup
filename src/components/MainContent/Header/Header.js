import React from "react";
import classes from "./Header.module.scss";
import moment from "moment";

const header = props => {
  if (props.isLoading) return null;
  console.log("props", JSON.stringify(props));
  return (
    <React.Fragment>
      <header className={classes.mainHeader}>
        <div className={classes.headerInfo}>
          <div>
            <h1 data-testid="h1tag">Weather in </h1>
            <span>
              {props.locationNameSelected}
              <span>   {`${props.longitudeLatitudeSelected[0]} N ${props.longitudeLatitudeSelected[1]} E`}</span>
            </span>
          </div>
          <div className={classes.headerImage}>
            <div
              title={
                props.isLoading ? props.current.weather[0].description : ""
              }
            >
              <img
                alt={
                  props.isLoading ? props.current.weather[0].description : ""
                }
                src={props.isLoading ? props.current.weather[0].icon : ""}
                style={{ height: "54px" }}
              />
              <span>{`${props.current.temp} Â°C`}</span>
            </div>
            <div>
            </div>
          </div>
        </div>
      </header>
    </React.Fragment>
  );
};

export default React.memo(header);
