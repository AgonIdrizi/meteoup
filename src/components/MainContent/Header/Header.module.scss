.mainHeader {
    height: 150px;
    position: relative;
    display:flex;
    justify-content: center;
    background-image: linear-gradient(135deg,#103f78,#0f8eb4 );
    background-size: cover;
    color: #fff;
} 
.headerInfo {
    display: flex;
    align-items: center;
    justify-content: space-between;
    width:100%;
    margin: 20px;
    
   
        
    text-align: 'left';

        
    
   
   h1{
       color:white;
       font-size: 2.6rem;
       margin-top: .2em;
       margin-bottom: 0px;
       font-weight: 600;
   }
   
}
.headerImage {
    margin-left: auto;
    margin-right: 20px;
    div {
      display: flex;
      justify-content: center
    }
    div:first-child {
        span{
            font-weight: 700;
            font-size: 2.6em
        }
    }

}

@media only screen and (max-width: 500px) {
    .headerInfo {
        flex-direction: column;
      h1 {
        font-size: 1.3rem;
        font-weight: 600;
      }
    }
    .headerImage {
        margin-left: 20px;
        div:first-child {
            display: flex;
            align-items: center;
            span{
              font-size: 1.3rem;
              font-weight: 600;
            }
        }  
    }
  }