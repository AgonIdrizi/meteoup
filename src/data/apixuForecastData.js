export const location = {
  name: "Tetovo",
  region: "Tetovo",
  country: "Macedonia",
  lat: 42.01,
  lon: 20.97,
  tz_id: "Europe/Skopje",
  localtime_epoch: 1561553535,
  localtime: "2019-06-26 14:52"
};
export const current = {
  last_updated_epoch: 1561553109,
  last_updated: "2019-06-26 14:45",
  temp_c: 31.0,
  temp_f: 87.8,
  is_day: 1,
  condition: {
    text: "Partly cloudy",
    icon: "//cdn.apixu.com/weather/64x64/day/116.png",
    code: 1003
  },
  wind_mph: 9.4,
  wind_kph: 15.1,
  wind_degree: 40,
  wind_dir: "NE",
  pressure_mb: 1021.0,
  pressure_in: 30.6,
  precip_mm: 0.9,
  precip_in: 0.04,
  humidity: 38,
  cloud: 50,
  feelslike_c: 31.1,
  feelslike_f: 88.1,
  vis_km: 10.0,
  vis_miles: 6.0,
  uv: 8.0,
  gust_mph: 10.5,
  gust_kph: 16.9
};
export const forecast = {
  forecastday: [
    {
      date: "2019-06-26",
      date_epoch: 1561507200,
      day: {
        maxtemp_c: 24.4,
        maxtemp_f: 75.9,
        mintemp_c: 14.3,
        mintemp_f: 57.7,
        avgtemp_c: 20.8,
        avgtemp_f: 69.4,
        maxwind_mph: 8.1,
        maxwind_kph: 13.0,
        totalprecip_mm: 8.5,
        totalprecip_in: 0.33,
        avgvis_km: 9.5,
        avgvis_miles: 5.0,
        avghumidity: 81.0,
        condition: {
          text: "Patchy rain possible",
          icon: "//cdn.apixu.com/weather/64x64/day/176.png",
          code: 1063
        },
        uv: 8.8
      },
      astro: {
        sunrise: "05:02 AM",
        sunset: "08:16 PM",
        moonrise: "01:11 AM",
        moonset: "01:34 PM"
      }
    },
    {
      date: "2019-06-27",
      date_epoch: 1561593600,
      day: {
        maxtemp_c: 27.1,
        maxtemp_f: 80.8,
        mintemp_c: 8.4,
        mintemp_f: 47.1,
        avgtemp_c: 20.8,
        avgtemp_f: 69.5,
        maxwind_mph: 4.9,
        maxwind_kph: 7.9,
        totalprecip_mm: 0.0,
        totalprecip_in: 0.0,
        avgvis_km: 10.0,
        avgvis_miles: 6.0,
        avghumidity: 73.0,
        condition: {
          text: "Partly cloudy",
          icon: "//cdn.apixu.com/weather/64x64/day/116.png",
          code: 1003
        },
        uv: 9.4
      },
      astro: {
        sunrise: "05:02 AM",
        sunset: "08:16 PM",
        moonrise: "01:36 AM",
        moonset: "02:34 PM"
      }
    },
    {
      date: "2019-06-28",
      date_epoch: 1561680000,
      day: {
        maxtemp_c: 25.7,
        maxtemp_f: 78.3,
        mintemp_c: 10.4,
        mintemp_f: 50.7,
        avgtemp_c: 19.0,
        avgtemp_f: 66.1,
        maxwind_mph: 5.4,
        maxwind_kph: 8.6,
        totalprecip_mm: 0.0,
        totalprecip_in: 0.0,
        avgvis_km: 10.0,
        avgvis_miles: 6.0,
        avghumidity: 70.0,
        condition: {
          text: "Partly cloudy",
          icon: "//cdn.apixu.com/weather/64x64/day/116.png",
          code: 1003
        },
        uv: 9.3
      },
      astro: {
        sunrise: "05:03 AM",
        sunset: "08:16 PM",
        moonrise: "02:02 AM",
        moonset: "03:38 PM"
      }
    },
    {
      date: "2019-06-29",
      date_epoch: 1561766400,
      day: {
        maxtemp_c: 22.3,
        maxtemp_f: 72.1,
        mintemp_c: 3.8,
        mintemp_f: 38.8,
        avgtemp_c: 14.4,
        avgtemp_f: 58.0,
        maxwind_mph: 5.1,
        maxwind_kph: 8.3,
        totalprecip_mm: 0.2,
        totalprecip_in: 0.01,
        avgvis_km: 10.0,
        avgvis_miles: 6.0,
        avghumidity: 72.0,
        condition: {
          text: "Patchy rain possible",
          icon: "//cdn.apixu.com/weather/64x64/day/176.png",
          code: 1063
        },
        uv: 9.0
      },
      astro: {
        sunrise: "05:03 AM",
        sunset: "08:16 PM",
        moonrise: "02:31 AM",
        moonset: "04:43 PM"
      }
    },
    {
      date: "2019-06-30",
      date_epoch: 1561852800,
      day: {
        maxtemp_c: 21.5,
        maxtemp_f: 70.7,
        mintemp_c: 7.7,
        mintemp_f: 45.9,
        avgtemp_c: 16.4,
        avgtemp_f: 61.5,
        maxwind_mph: 4.9,
        maxwind_kph: 7.9,
        totalprecip_mm: 0.0,
        totalprecip_in: 0.0,
        avgvis_km: 10.0,
        avgvis_miles: 6.0,
        avghumidity: 70.0,
        condition: {
          text: "Partly cloudy",
          icon: "//cdn.apixu.com/weather/64x64/day/116.png",
          code: 1003
        },
        uv: 9.2
      },
      astro: {
        sunrise: "05:04 AM",
        sunset: "08:16 PM",
        moonrise: "03:06 AM",
        moonset: "05:51 PM"
      }
    },
    {
      date: "2019-07-01",
      date_epoch: 1561939200,
      day: {
        maxtemp_c: 23.8,
        maxtemp_f: 74.8,
        mintemp_c: 4.7,
        mintemp_f: 40.5,
        avgtemp_c: 14.8,
        avgtemp_f: 58.7,
        maxwind_mph: 4.3,
        maxwind_kph: 6.8,
        totalprecip_mm: 0.0,
        totalprecip_in: 0.0,
        avgvis_km: 10.0,
        avgvis_miles: 6.0,
        avghumidity: 71.0,
        condition: {
          text: "Partly cloudy",
          icon: "//cdn.apixu.com/weather/64x64/day/116.png",
          code: 1003
        },
        uv: 4.0
      },
      astro: {
        sunrise: "05:04 AM",
        sunset: "08:15 PM",
        moonrise: "03:47 AM",
        moonset: "06:58 PM"
      }
    },
    {
      date: "2019-07-02",
      date_epoch: 1562025600,
      day: {
        maxtemp_c: 28.3,
        maxtemp_f: 82.9,
        mintemp_c: 8.9,
        mintemp_f: 48.0,
        avgtemp_c: 19.3,
        avgtemp_f: 66.7,
        maxwind_mph: 4.3,
        maxwind_kph: 6.8,
        totalprecip_mm: 0.7,
        totalprecip_in: 0.03,
        avgvis_km: 10.0,
        avgvis_miles: 6.0,
        avghumidity: 63.0,
        condition: {
          text: "Patchy rain possible",
          icon: "//cdn.apixu.com/weather/64x64/day/176.png",
          code: 1063
        },
        uv: 4.0
      },
      astro: {
        sunrise: "05:05 AM",
        sunset: "08:15 PM",
        moonrise: "04:35 AM",
        moonset: "08:03 PM"
      }
    }
  ]
};
