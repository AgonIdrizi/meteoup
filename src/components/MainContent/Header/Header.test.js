import React from "react";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Header from "./Header";
import { location, current } from "../../../data/apixuForecastData";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(
    <Header location={location} current={current} />
  );
  expect(asFragment()).toMatchSnapshot();
});

it("inserts text in h1", () => {
  const { getByTestId, getByText } = render(
    <Header location={location} current={current} />
  );
  expect(getByTestId("h1tag")).toHaveTextContent("Weather in Tetovo");
});

it("displays temp", () => {
  const { container } = render(
    <Header location={location} current={current} />
  );
  const headerElem = container.getElementsByClassName("headerImage");
  expect(headerElem[0]).toHaveTextContent("31 Â°C");
});
