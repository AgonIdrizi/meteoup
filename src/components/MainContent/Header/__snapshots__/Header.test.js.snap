// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <header
    class="mainHeader"
  >
    <div
      class="headerInfo"
    >
      <div>
        <h1
          data-testid="h1tag"
        >
          Weather in Tetovo 
        </h1>
        <span>
          Tetovo, Macedonia, 
          <span>
            42.01 N 20.97 E
          </span>
        </span>
      </div>
      <div
        class="headerImage"
      >
        <div
          title="Partly cloudy"
        >
          <img
            alt="Partly cloudy"
            src="//cdn.apixu.com/weather/64x64/day/116.png"
            style="height: 54px;"
          />
          <span>
            31 Â°C
          </span>
        </div>
        <div>
          <span>
            Last Updated a month ago
          </span>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
